File: code/method_block_passing.py
def my_select(arr):
    yield from arr

print([x for x in my_select([1, 5, 10]) if x < 6])


def my_select(arr, filter):
    for a in arr:
        if filter(a):
            yield a

print(list(my_select([1, 5, 10], lambda x: x < 6)))

File: code/ruby/method_block_passing.rb
def my_select(arr)
  selected = []
  arr.each do |a|
    selected.push a if yield(a)
  end
  selected
end
puts my_select [1, 5, 10] { |x| x < 6 }


def my_select(arr, &filter)
  selected = []
  arr.each do |a|
    selected.push a if filter.call(a)
  end
  selected
end
puts my_select [1, 5, 10] { |x| x < 6 }
